core/src/main/kotlin/org/evomaster/core/search/gene/StringGene.kt

```diff
@@ -155,8 +155,13 @@ class StringGene(
         assert(specializations.isNotEmpty())
 
         specializationGene = when {
+
             specializations.any { it.stringSpecialization == DATE_YYYY_MM_DD } -> DateGene(name)
 
+            specializations.any { it.stringSpecialization == DATE_YYYY_MM_DD_HH_SS } -> DateTimeGene(name)
+
```

Comment:
is this correct? `DateTimeGene` is currently based on RFC3339, where there is a "T" between the date and the time. might need to extend those genes by providing possible different patterns when getting the phenotype out of the genotype. 

anyway, i m refactoring this class, so i ll do the actual integration here

Reply:
ok. just in case I adapted the code to your last changes

Reply:
I did not add the code for handling DATE_YYYY_MM_DD_HH_SS, DATE_FORMAT_PATTERN and DATE_FORMAT_UNKNOWN_PATTERN